crates/smdk/src/build.rs

```diff
@@ -0,0 +1,32 @@
+use anyhow::{Error, Result};
+use clap::Parser;
+use std::process::Command;
+use std::fmt::Debug;
+
+/// Builds the SmartModule in the current working directory into a WASM file
+#[derive(Debug, Parser)]
+pub struct BuildOpt;
+
+impl BuildOpt {
+    pub(crate) fn process(&self) -> Result<()> {
+        let mut cmd = Command::new("cargo");
```

Comment:
maybe we should check that `cargo` is installed first

Reply:
Sure! Now I'm checking `cargo version` and catching any error while running that before going straight to the `cargo build` cmd.

This way if something happens running `cargo version` we acknowledge there's an issue with `cargo` binary instead of the build.

Reply:
or other ways it to catch exception if it doesn't exist and display proper error message